aip/general/0122.md

```diff
@@ -102,17 +110,26 @@ ID for the publisher, and `les-miserables` is the resource ID for the book.
   server-generated if unset), or never set by users (not accepted at resource
   creation). They **should** be immutable once created.
   - If resource IDs are user-settable, the API **must** document allowed
-    formats.
+    formats. User-settable resource IDs **should** conform to [RFC-1034][],
+    which restricts to letters, numbers, and hyphen, with a 63 character
+    maximum.
+    - Additionally, user-settable resource IDs **should** restrict letters to
+      lower-case.
+    - Characters outside of ASCII **should not** be permitted; however, if
+      Unicode characters are necessary, APIs **must** follow guidance in
+      AIP-210.
+    - User-settable IDs **should not** be permitted to be a UUID (or any value
+      that syntactically appears to be a UUID).
   - If resource IDs are not user-settable, the API **should** document the
-    basic format, and any upper boundaries (for example, "at most 256
+    basic format, and any upper boundaries (for example, "at most 63
     characters").
   - For more information, see the [create][] standard method.
-- Resource IDs **should** only use characters that do not require URL-escaping.
-  - Characters outside of ASCII are discouraged; however, if Unicode characters
-    are necessary, APIs **must** follow guidance in [AIP-210][].
+
+**Important:** Resources that are declarative-friendly (AIP-128) **must** use
```

Comment:
And must implement the "not permitted to be a UUID" rule? (What about other rules? Maybe I should wait until I've reviewed 128...)

Reply:
I think that is already covered by the bullets above.